src/state/migrations/2.test.ts

```diff
@@ -0,0 +1,62 @@
+import { createMigrate } from 'redux-persist'
+import { RouterPreference } from 'state/routing/types'
+import { SlippageTolerance } from 'state/user/types'
+
+import { migration1 } from './1'
+import { migration2, PersistAppStateV2 } from './2'
+
+const previousState: PersistAppStateV2 = {
+  user: {
+    userLocale: null,
+    // @ts-ignore this is intentionally a string and not the `RouterPreference` enum because `client` is a deprecated option
+    userRouterPreference: 'client',
+    userHideClosedPositions: false,
+    userSlippageTolerance: SlippageTolerance.Auto,
+    userSlippageToleranceHasBeenMigratedToAuto: true,
+    userDeadline: 1800,
+    tokens: {},
+    pairs: {},
+    timestamp: Date.now(),
+    hideBaseWalletBanner: false,
+  },
+  _persist: {
+    version: 1,
+    rehydrated: true,
+  },
+}
+
+describe('migration to v2', () => {
+  it('should migrate users who currently have `client` router preference', async () => {
+    const migrator = createMigrate(
+      {
+        1: migration1,
+        2: migration2,
```

Comment:
qq: do you need all previous migrations in the migrator or can you just use the relevant one ie 2?

Reply:
technically only need `2` but its more similar to the behavior on prod if you pass in all :D